packages/client/src/links/internals/httpUtils.ts

```diff
@@ -153,6 +155,38 @@ export const jsonHttpRequester: Requester = (opts) => {
   });
 };
 
+export const universalRequester: Requester = (opts) => {
+  const input = getInput(opts);
+  if (isFormData(input)) {
+    if (opts.type !== 'mutation') {
+      throw new Error('FormData is only supported for mutations');
+    }
+
+    return httpRequest({
+      ...opts,
+      // The browser will set this automatically and include the boundary= in it
+      contentTypeHeader: undefined,
+      getUrl,
+      getBody: () => input,
+    });
+  }
+
+  if (isOctetType(input)) {
+    if (opts.type !== 'mutation') {
```

Comment:
Did we ship the method override option yet? In that case this would be dependent on the http method rather than procedure type, right?

Reply:
You're quite possible right, but I also wouldn't expect a binary file from a query. Might be an area to refine a bit since I believe there's a lot of these checks we need to consider now

Reply:
Yeah I'm fine that you can method override and send binary data in queries, but @juliusmarminge is right that we "should" allow it if it's method overridden

Reply:
Changed to this:

```
    if (opts.type !== 'mutation' || opts.methodOverride === 'POST') {
      throw new Error('FormData is only supported for mutations');
    }
```